didman/didman.go

```diff
@@ -312,23 +318,32 @@ func (d *didman) SearchOrganizations(ctx context.Context, query string, didServi
 	// Convert organization concepts and DID documents to search results
 	results := make([]OrganizationSearchResult, len(organizations))
 	for i := range organizations {
-		organization, ok := organizations[i]["organization"].(map[string]interface{})
-		if !ok {
-			return nil, errors.New("unable to map organization to concept")
+		reader := jsonld.Reader{DocumentLoader: d.contextManager.DocumentLoader()}
+		document, err := reader.Read(organizations[i])
+		if err != nil {
+			return nil, fmt.Errorf("failed to expand credential to JSON-LD: %w", err)
 		}
+
+		// guaranteed to contain values
+		orgNames := document.ValueAt(jsonld.NewPath(jsonld.OrganizationNamePath...))
```

Comment:
Maybe make `jsonld.OrganizationNamePath` already of type `Path`, to deduplicate the `NewPath` on constant values (there's some duplication on that part in this PR that can easily be removed)